Dockerfile

```diff
@@ -0,0 +1,89 @@
+# syntax=docker/dockerfile:1
+
+#	This Source Code Form is subject to the terms of the Mozilla Public License, 
+#	v. 2.0. If a copy of the MPL was not distributed with this file, You can 
+#	obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under 
+#	the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+#	
+#	Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS 
+#	graphic logo is a trademark of OpenMRS Inc.
+
+### Build Stage
+FROM maven:3.8-jdk-11 as build
+WORKDIR /app
+
+# Copy source
+COPY . .
+
+# Build the app using maven cache
+RUN --mount=type=cache,target=/root/.m2 mvn clean package -DskipTests
```

Comment:
I'm surprised we are building the war during Docker image creation.  I would have thought we would use pre-built artifacts.  Ideally we'd have 1 official version of each build (whether snapshot or release) in our Maven repository, and those versions are what would be bundled.  Then there are no surprises.

Reply:
I'm definitely on board with moving all of our CI builds to use Docker.  That's something I've been working on personally in the PIH CI.  But that's not what we are talking about here, is it?  This container is about running OpenMRS in Tomcat, not building, testing, and deploying (otherwise you wouldn't be skipping tests below, and you'd be deploying to maven, etc).  Plus, there are a lot more things than OpenMRS core that we need to reliably build, test, and deploy to Maven, and we likely want some base images that strictly deal with building and deploying.  There are nuances there like how to setup Maven credentials, SSH and GPG keys, and run Selenium, Junit, and Jasmine tests in some cases.  What this seems to be doing is just building OpenMRS core because it is more convenient than downloading from Maven.

Reply:
If you look at the Dockerfile it has 2 stages. First is the build stage, which is responsible for building and testing openmrs-core. It will be also used for pushing to maven repo and releasing. Tests are indeed skipped here for now, but I will enable them and make it possible to pass maven parameters by devs to skip them. Second is the runtime stage, which is building the tomcat image. It uses the results of the build stage to create the runtime environment (runtime image). The runtime image has none of the dev dependencies (not even maven). It is the one that will be pushed to dockerhub. The build stage and resulting image will be used be devs and CI only.

Reply:
That seems cool.  These stages are not something I'm familiar with.  Need to do some reading up.  Thanks!